dist/ruff_check.py

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env python3
+
+import pathlib
+import re
+from shutil import which
+import subprocess
+import sys
+
+# Always use this scripts folder as the working directory.
+current_dir = pathlib.Path(__file__).parent.resolve()
+ruff_config = f"{current_dir}/ruff.toml"
+
+
+def run(cmd):
+    try:
+        output = subprocess.check_output(cmd, cwd=current_dir).decode().strip()
+        if output == "All checks passed!":
+            return True
+        else:
+            print(output)
+            return False
+    except subprocess.CalledProcessError as cpe:
+        print("The command [%s] failed:\n%s" % (" ".join(cmd), cpe.output.decode("utf-8")))
+        return False
+
+
+if not which("ruff"):
+    doc_link = "https://docs.astral.sh/ruff/installation/"
+    lines = []
+    ruff_version = None
+    with open(ruff_config) as file:
+        for line in file:
+            if m := re.search(r'required-version = "==(\d.\d.\d)"', line.strip()):
+                ruff_version = m.group(1)
+                break
+    print("Ruff is not installed!")
+    print("\nCreate a virtual environment if you haven't yet:")
```

Comment:
Is this the first time we're mandating a virtualenv in order to run `s_all`? I've never needed one in the past.

Reply:
I am happy to change the wording here. It's really optional, everyone is free to manage their local packages the way they want.

Reply:
@pmacko86 suggested that `s_all` would create its own env, which is a possible improvement that we can do in a different ticket.

Reply:
SGTM. I'd prefer not mentioned virtualenv in the comment at the moment and have a generic "Please install ruff"